bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccHandler.java

```diff
@@ -0,0 +1,443 @@
+                return;
+            switch (channelIdWithoutGroup) {
+                case CHANNEL_LOADPOINT_MODE:
+                    evccAPI.setMode(loadpoint, command.toString());
+                    break;
+                case CHANNEL_LOADPOINT_MIN_SOC:
+                    evccAPI.setMinSoC(loadpoint, Integer.parseInt(command.toString().replaceAll(" %", "")));
+                    break;
+                case CHANNEL_LOADPOINT_TARGET_SOC:
+                    evccAPI.setTargetSoC(loadpoint, Integer.parseInt(command.toString().replaceAll(" %", "")));
+                    break;
+                case CHANNEL_LOADPOINT_TARGET_TIME:
+                    if (targetTimeEnabled == true) {
+                        targetTimeZDT = new DateTimeType(command.toString()).getZonedDateTime();
+                        evccAPI.setTargetCharge(loadpoint, targetSoC, targetTimeZDT);
+                        ChannelUID channel = new ChannelUID(getThing().getUID(), "loadpoint" + loadpoint,
+                                CHANNEL_LOADPOINT_TARGET_TIME);
+                        updateState(channel, new DateTimeType(targetTimeZDT));
+                    }
+                    break;
+                case CHANNEL_LOADPOINT_TARGET_TIME_ENABLED:
+                    if (command == OnOffType.ON) {
+                        targetTimeEnabled = true;
+                        evccAPI.setTargetCharge(loadpoint, targetSoC, targetTimeZDT);
+                    } else {
+                        targetTimeEnabled = false;
+                        evccAPI.unsetTargetCharge(loadpoint);
+                    }
+                    break;
+                case CHANNEL_LOADPOINT_PHASES:
+                    evccAPI.setPhases(loadpoint, Integer.parseInt(command.toString()));
+                    break;
+                case CHANNEL_LOADPOINT_MIN_CURRENT:
+                    evccAPI.setMinCurrent(loadpoint, Integer.parseInt(command.toString().replaceAll(" A", "")));
+                    break;
+                case CHANNEL_LOADPOINT_MAX_CURRENT:
+                    evccAPI.setMaxCurrent(loadpoint, Integer.parseInt(command.toString().replaceAll(" A", "")));
+                    break;
+                default:
+                    return;
+            }
+            refresh();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        this.config = getConfigAs(EvccConfiguration.class);
+        EvccConfiguration config = this.config;
+        if (config != null) {
+            String url = config.url;
+            if (url == null || "".equals(url)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                        "@text/offline.configuration-error.no-host");
+                return;
+            }
+            this.evccAPI = new EvccAPI(url);
+            logger.debug("Setting up refresh job ...");
+            statePollingJob = scheduler.scheduleWithFixedDelay(this::refresh, 0, 15, TimeUnit.SECONDS);
```

Comment:
Should the value 15 here be replaced by **refreshInterval** from the configuration?

Reply:
Oops, yes the refresh interval from the config should be there.